libs/sdk-core/src/swap_out/reverseswap.rs

```diff
@@ -170,9 +170,17 @@ impl BTCSendSwap {
         // The result of the creation call can succeed or fail
         // We update the rev swap status accordingly, which would otherwise have needed a fully fledged sync() call
         match res {
-            Ok(_) => self
-                .persister
-                .update_reverse_swap_status(&created_rsi.id, &InProgress)?,
+            Ok(_) => {
+                let lockup_txid = self
+                    .get_lockup_tx(&created_rsi)
+                    .await?
+                    .map(|lockup_tx| lockup_tx.txid)
+                    .unwrap_or_default();
+                self.persister
+                    .update_reverse_swap_status(&created_rsi.id, &InProgress)?;
+                self.persister
+                    .update_reverse_swap_lockup_txid(&created_rsi.id, &lockup_txid)?
```

Comment:
At this point the lockup_txid could still potentially be "" if it's not been seen by the mempool after the reverse swap in created? (tell me if I'm wrong). I think it's preferable if the lockup_txid column is only set if it's a valid lockup tx.

Reply:
> Do we still need to look at BreezServices send_onchain and in_progress_reverse_swaps? The swapper's convert_reverse_swap_info method is called on each swap, in turn calling get_lockup_tx and create_claim_tx (when applicable). Could we simply use the cached txid's here?

The idea would be not to change these other flows so that they always get up-to-date information. For the purpose of `list_payments`, it isn't as important that the information is synced. WDYT?

> At this point the lockup_txid could still potentially be "" if it's not been seen by the mempool after the reverse swap in created? (tell me if I'm wrong). I think it's preferable if the lockup_txid column is only set if it's a valid lockup tx.

True. Addressed in [895fad2](https://github.com/breez/breez-sdk/pull/728/commits/895fad29208a86f31ffa4ad846666893df63fe65).